icu4c/source/test/intltest/msfmrgts.cpp

```diff
@@ -987,11 +988,29 @@ void MessageFormatRegressionTest::Test4112104()
     delete format;
 }
 
+void MessageFormatRegressionTest::TestICU12584() {
+    // 42 placeholders.
+    // We were able to handle only 10 due to the bug ICU-12584.
+     UnicodeString pattern("{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}" \
+        "{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}" \
+        "{3}{1}{2}{3}");
+
+    UErrorCode status = U_ZERO_ERROR;
+    int count;
+    const Format **formats;
```

Comment:
Please also move variable declarations close to their first use, such as below
```
int32_t count;
const Format **formats = msg.getFormats(count);
```

Reply:
Done